server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -1002,55 +1008,74 @@ private void findNodes(MNode node, PartialPath path, List<PartialPath> res, int
   }
 
   public void serializeTo(String snapshotPath) throws IOException {
-    try (BufferedWriter bw = new BufferedWriter(
-        new FileWriter(SystemFileFactory.INSTANCE.getFile(snapshotPath)))) {
-      root.serializeTo(bw);
+    MLogWriter mLogWriter = null;
+    try {
+      mLogWriter = new MLogWriter(snapshotPath);
+      root.serializeTo(mLogWriter);
+    } finally {
+      if (mLogWriter != null) {
+        mLogWriter.close();
+      }
     }
   }
 
   @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
   public static MTree deserializeFrom(File mtreeSnapshot) {
-    try (BufferedReader br = new BufferedReader(new FileReader(mtreeSnapshot))) {
-      String s;
+    MLogReader mlogReader = null;
+    try {
```

Comment:
Use try with resource to avoid new sonar code smell..

Reply:
Got it!